tests/python_package_test/test_dask.py

```diff
@@ -415,6 +451,165 @@ def test_ranker(output, client, listen_port, group):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+@pytest.mark.parametrize('task', ['classification', 'regression', 'ranking'])
+def test_training_works_if_client_not_provided_or_set_after_construction(task, listen_port, client):
+    if task == 'ranking':
+        _, _, _, _, dX, dy, _, dg = _create_ranking_data(
+            output='array',
+            group=None
+        )
+        model_factory = lgb.DaskLGBMRanker
+    else:
+        _, _, _, dX, dy, _ = _create_data(
+            objective=task,
+            output='array',
+        )
+        dg = None
+        if task == 'classification':
+            model_factory = lgb.DaskLGBMClassifier
+        elif task == 'regression':
+            model_factory = lgb.DaskLGBMRegressor
+
+    params = {
+        "time_out": 5,
+        "local_listen_port": listen_port,
+        "n_estimators": 1,
+        "num_leaves": 2
+    }
+
+    # fit should work if client isn't provided
+    dask_model = model_factory(**params)
+    assert dask_model._client is None
+    assert dask_model.client == client
+
+    dask_model.fit(dX, dy, group=dg)
+    assert dask_model.fitted_
+    assert dask_model._client is None
+    assert dask_model.client == client
+
+    preds = dask_model.predict(dX)
+    assert isinstance(preds, da.Array)
+    assert dask_model.fitted_
+    assert dask_model._client is None
+    assert dask_model.client == client
+
+    local_model = dask_model.to_local()
+    assert local_model._client is None
+    with pytest.raises(AttributeError):
+        local_model.client
+
+    # should be able to set client after construction
+    dask_model = model_factory(**params)
+    dask_model.client = client
+    assert dask_model._client == client
+    assert dask_model.client == client
+
+    dask_model.fit(dX, dy, group=dg)
+    assert dask_model.fitted_
+    assert dask_model._client == client
+    assert dask_model.client == client
+
+    preds = dask_model.predict(dX)
+    assert isinstance(preds, da.Array)
+    assert dask_model.fitted_
+    assert dask_model._client == client
+    assert dask_model.client == client
+
+    local_model = dask_model.to_local()
+    assert local_model._client is None
+    with pytest.raises(AttributeError):
+        local_model.client
+
+    client.close(timeout=CLIENT_CLOSE_TIMEOUT)
+
+
+@pytest.mark.parametrize('serializer', ['pickle', 'joblib', 'cloudpickle'])
```

Comment:
I chose to test all three of these because:

* `pickle` is part of the standard library, and very popular
* `joblib` is popular with `scikit-learn`'s ecosystem, and tested in LightGBM's sklearn tests: https://github.com/microsoft/LightGBM/blob/ea8e47ea24fb942155f16da6720b32de976d5731/tests/python_package_test/test_sklearn.py#L393
* `cloudpickle` is the preferred alternative to `pickle` in the Dask ecosystem
    - `distributed` already has a strong dependency on it (https://github.com/dask/distributed/blob/08ea96890674d48b90f4e1f92959957e5e362a18/requirements.txt#L2), so this doesn't introduce any new dependencies for LightGBM's tests and any users of the Dask module will already have `cloudpickle`